habitat-lab/habitat/robots/robot_base.py

```diff
@@ -145,14 +225,113 @@ def is_base_link(self, link_id: int) -> bool:
             self.sim_obj.get_link_name(link_id) in self.params.base_link_names
         )
 
-    def _update_motor_settings_cache(self):
-        """Updates the JointMotorSettings cache for cheaper future updates"""
-        self.joint_motors = {}
-        for (
-            motor_id,
-            joint_id,
-        ) in self.sim_obj.existing_joint_motor_ids.items():
-            self.joint_motors[joint_id] = (
-                motor_id,
-                self.sim_obj.get_joint_motor_settings(motor_id),
+    def _validate_ctrl_input(self, ctrl: List[float], joints: List[int]):
+        """
+        Raises an exception if the control input is NaN or does not match the
+        joint dimensions.
+        """
+        if len(ctrl) != len(joints):
+            raise ValueError(
+                f"Control dimension does not match joint dimension: {len(ctrl)} vs {len(joints)}"
+            )
+        if np.any(np.isnan(ctrl)):
+            raise ValueError("Control is NaN")
+
+    def update_base(self, rigid_state, target_rigid_state):
+
+        cur_state = self._sim.robot.sim_obj.transformation
+
+        # Conduct the collision detection
+        has_attr_contact_test = hasattr(
+            self._sim.habitat_config, "contact_test"
+        )
+        has_attr_navmesh = hasattr(self._sim.habitat_config, "nav_mesh")
+        if (
+            has_attr_contact_test and self._sim.habitat_config.contact_test
+        ) and (
+            (has_attr_navmesh and not self._sim.habitat_config.nav_mesh)
+            or (not has_attr_navmesh)
+        ):
+            did_collide = self._sim.contact_test(self.sim_obj.object_id)
+            robot_id = self.sim_obj.object_id
+            end_pos = target_rigid_state.translation
+            # If there is a collision, we do projection
```

Comment:
I'm not sure this is tested enough yet. For example, there is no validation that the projected position is collision free. It could be that projection from initial collisions causes new collisions.

Maybe consider removing the projection code for now and introducing that as a separate PR with tests.

Reply:
Remove the projection code here. thank you!